src/modules/roc_pipeline/converter_source.cpp

```diff
@@ -35,16 +35,18 @@ ConverterSource::ConverterSource(const ConverterConfig& config,
 
         resampler_.reset(audio::ResamplerMap::instance().new_resampler(
                              config.resampler_backend, allocator, config.resampler,
-                             config.input_channels, config.internal_frame_size),
+                             config.internal_frame_length, config.input_sample_rate,
+                             config.input_channels),
                          allocator);
 
         if (!resampler_) {
             return;
         }
 
-        resampler_reader_.reset(new (allocator)
-                                    audio::ResamplerReader(*areader, *resampler_, pool,
-                                                           config.internal_frame_size),
+        resampler_reader_.reset(new (allocator) audio::ResamplerReader(
+                                    *areader, *resampler_, pool,
+                                    config.internal_frame_length,
+                                    config.input_sample_rate, config.input_channels),
```

Comment:
I think ResamplerReader should use output_sample_rate & output_channels instead of input_sample_rate & input_channels as we splitting the output stream?

Reply:
> ResamplerReader uses frame_length to split its input stream into frames. So we should pass to it the sample rate of the input stream, not of the output stream. 
Saw this comment only now. So I think I will do a similar thing here.

Reply:
> So I think ResamplerReader should use input_sample_rate.

Yes.